lib/utilities/MD5Sum.js

```diff
@@ -0,0 +1,24 @@
+const Transform = require('stream').Transform;
+const crypto = require('crypto');
+
+class MD5Sum extends Transform {
+
+    constructor(callback) {
+        super({});
+        this.hash = crypto.createHash('md5');
+        this.callback = callback;
+    }
+
+    _transform(chunk, encoding, callback) {
+        this.hash.update(chunk, encoding);
+        callback(null, chunk, encoding);
+    }
+
+    _flush(callback) {
```

Comment:
the done callback is now sending to the constructor, and callback is used as usual in functions :)